File: Actito.Inbox.Core/Models/ActitoInboxItem.cs
using ActitoSdk.Core.Models;

namespace ActitoSdk.Inbox.Core.Models;

/// <summary>
/// Represents an item in the Actito inbox.
/// </summary>
/// <remarks>
/// An <see cref="ActitoInboxItem"/> contains a notification and metadata about its read
/// state within the inbox. Inbox items can optionally have an expiration date.
/// </remarks>
public class ActitoInboxItem
{
    /// <summary>
    /// Unique identifier of the inbox item.
    /// </summary>
    public string Id { get; }

    /// <summary>
    /// <see cref="ActitoNotification"/> associated with this inbox item.
    /// </summary>
    public ActitoNotification Notification { get; }

    /// <summary>
    /// Timestamp indicating when the item was received.
    /// </summary>
    public DateTime Time { get; }

    /// <summary>
    /// Indicates whether the item has been opened by the user.
    /// </summary>
    public bool Opened { get; }

    /// <summary>
    /// Optional expiration timestamp of the item.
    /// </summary>
    public DateTime? Expires { get; }

    /// <summary>
    /// Constructor for <see cref="ActitoInboxItem"/>.
    /// </summary>
    public ActitoInboxItem(string id, ActitoNotification notification, DateTime time, bool opened,
        DateTime? expires)
    {
        Id = id;
        Notification = notification;
        Time = time;
        Opened = opened;
        Expires = expires;
    }
}

File: Actito.Inbox.iOS.Binding.Bridge/Sources/Models/ActitoInboxItem.swift
import Foundation
import ActitoInboxKit
import ActitoBinding

@objc
public class ActitoInboxItem : NSObject {
    @objc public let inboxItemId: String
    @objc public let notification: ActitoNotification
    @objc public let time: Date
    @objc public let opened: Bool
    @objc public let expires: Date?

    @objc public init(inboxItemId: String, notification: ActitoNotification, time: Date, opened: Bool, expires: Date?) {
        self.inboxItemId = inboxItemId
        self.notification = notification
        self.time = time
        self.opened = opened
        self.expires = expires
    }

    public convenience init(from item: ActitoInboxKit.ActitoInboxItem) {
        self.init(
            inboxItemId: item.id,
            notification: ActitoBinding.ActitoNotification(from: item.notification),
            time: item.time,
            opened: item.opened,
            expires: item.expires
        )
    }

    public func toNative() -> ActitoInboxKit.ActitoInboxItem {
        return ActitoInboxKit.ActitoInboxItem(
            id: inboxItemId,
            notification: notification.toNative(),
            time: time,
            opened: opened,
            expires: expires
        )
    }
}
